lib/fastifySession.js

```diff
@@ -15,8 +15,14 @@ function session (fastify, options, next) {
 
   options = ensureDefaults(options)
 
-  fastify.decorate('decryptSession', (sessionId, request, callback) => {
-    decryptSession(sessionId, options, request, callback)
+  fastify.decorate('decryptSession', (sessionId, request, cookieOpts, callback) => {
+    if (typeof cookieOpts === 'function') {
```

Comment:
Could you please add a comment why this is done this way? Mention the deprecation of the method.